packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -362,4 +343,300 @@ export class UiControlClient extends ApiCommands {
+   */
+  async typeIntoTextfield(
+    params: {
+      textToWrite: string,
+      relation: {
+        type?: Relations,
+        label: string
+      }
+    },
+  ) {
+    const command = this.typeIn(params.textToWrite).textfield();
+
+    if (params.relation.type) {
+      switch (params.relation.type) {
+        case 'nearestTo':
+          await command.nearestTo().text(params.relation.label).exec();
+          break;
+        case 'contains':
+          await command.contains().text(params.relation.label).exec();
+          break;
+        case 'leftOf':
+          await command.leftOf().text(params.relation.label).exec();
+          break;
+        case 'above':
+          await command.above().text(params.relation.label).exec();
+          break;
+        case 'rightOf':
+          await command.rightOf().text(params.relation.label).exec();
+          break;
+        case 'below':
+          await command.below().text(params.relation.label).exec();
+          break;
+        default:
+          await command.nearestTo().text(params.relation.label).exec();
+          break;
+      }
+    } else {
+      await command.nearestTo().text(params.relation.label).exec();
+    }
+  }
+
+  /**
+   * Click on a specific text.
+   * You can also use a RegEx or match the text exactly by specifyicing the specific flag.
+   * Use a relation to find the text in relation to a specific text.
+   *
+   * **Examples:**
+   * ```typescript
+   * // Click text that matches exactly
+   * await aui.clickText({text: 'askui', exact: true})
+   *
+   * // Click text that contains 'pie' or 'cake' or 'Pie' or 'Cake'
+   * await aui.clickText({text: '.*([Pp]ie|[Cc]ake).*', regex: true})
+   *
+   * // Click the text 'TERMINAL' that is left of the text 'Ports'
+   * await aui.clickText({text: 'TERMINAL', relation: {type: 'leftOf', text: 'PORTS'}})
+   * ```
+   *
+   * @param {Object} params - Object containing required `text` property and optional properties
+   *                          for regular expression matching and relation.
+   * @property {string} params.text - The text to be clicked.
+   * @property {boolean} [params.regex=false] - Whether the text is matched using a regular
+   *                                            expression, default is false.
+   * @property {boolean} [params.exact=false] - Whether an exact match of the text is required
+   *                                            when using regex, default is false.
+   * @property {Object} [params.relation] - Object describing the relationship between the
+   *                                        clicked text and another element.
+   * @property {Relations} params.relation.type - The type of relation.
+   * @property {string} params.relation.text - The label or text associated with the
+   *                                           related element or state.
+   */
+  async clickText(
+    params: {
+      text: string,
+      regex?: boolean,
+      exact?: boolean,
```

Comment:
I think that these both correlate with each other can lead to confusing edge cases and increase complexity, e.g., what happens if `regex` and `exact` are both set to `true`. What about combining these two into one string union type, e.g., `type: 'similar' | 'exact' | 'regex' = 'similar'` or `type?: 'exact' | 'regex'`?